powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/cache/CacheManager.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.parameters.cache;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+public class CacheManager {
+    static final Duration DEFAULT_MAX_AGE_SECS = Duration.of(5, SECONDS);
+
+    private final DataStore store;
+    private final NowProvider nowProvider;
+    private Duration defaultMaxAge = DEFAULT_MAX_AGE_SECS;
+    private Duration maxAge = defaultMaxAge;
+
+    public CacheManager(NowProvider nowProvider) {
```

Comment:
Instead of having NowProvider can these classes depend on Clock instead ? That is the java8 recommended way of  dealing with manipulation of time

Reply:
OK, I did that. Looks better now.